Javascript challenge files

henyspider opened the pull request.

Commit:
adding the javascript challenge files

pgrobban commented on index.js:
```diff
@@ -0,0 +1,83 @@
+import fs from 'fs';
+
+const data = fs.readFileSync('words_alpha.txt', 'utf8');
+//console.log(data);
```
remove comments for outdated code.

pgrobban commented on index.js:
```diff
@@ -0,0 +1,83 @@
+    let result;
+    let foundWords = []; //This is the array that'll list the words that got their letters in alphabetical order
+
+    for(let i = 0; i < arr.length; i++){ // We loop through the words inside the array
```
check the spaces in the lines. for example there should be a space after for and if

pgrobban commented on index.js:
```diff
@@ -0,0 +1,83 @@
+
+const ALPHABET = ['a', 'b', 'c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z'];
+
+//console.log(isAlphabetical("acbde"))
```
remove comment

pgrobban commented on index.js:
```diff
@@ -0,0 +1,83 @@
+    return foundWords
+}
+
+function focusOnLetters(word){ //This function focuses on each word of the array
```
you don't need many of the comments here. the function and variable names should give a good idea what the code does. try to use comments only for the "why" not the "how"

pgrobban commented on index.js:
```diff
@@ -0,0 +1,83 @@
+    let takeLetters = word.split(""); //We're splitting up the word, and we get a new array with that word's letters as elements
+    let join;//We'll see the use of this on line 50
+
+    for(let j = 0; j < takeLetters.length; j++){//We're looping through the word's letters
```
space after for. remove the next line empty

pgrobban commented on index.js:
```diff
@@ -0,0 +1,83 @@
+
+    let count = 0; //The use of this count is explained line 65
+
+    for( let i = 0; i < arr.length-1; i++){//for each number inside the inOrder array, if they're in numerical order, add 1 to the count
```
space after for, not inside the parentheses

pgrobban commented on index.js:
```diff
@@ -0,0 +1,83 @@
+        /*If the count is equal to the length of the array, minus one (because [0, 1, 2, 3, 4] has 5 values but we're doing the calculation 5-1 times),
+        then all elements of the array have been checked, and they all passed, so the word has its letters in alphabetical order since
+        the letters indexes are all in order numerically*/
+        return 'in Alphabetical Order'
```
return boolean values instead.

pgrobban commented on index.js:
```diff
@@ -0,0 +1,83 @@
+    return join
+}
+
+const machination = arr => {// This function compares the numbers inside the inOrder Array
```
call this function isAlphabetic or something that makes it clear what its purpose is. i  have no idea what machination means.

pgrobban commented on index.js:
```diff
@@ -0,0 +1,83 @@
+    return 'not in Alphabetical Order'
+}
+
+console.log(focusOnWords(words))
```
this log can stay for now. but typically you wouldn't have these.

henyspider committed:
Update index.js

henyspider committed:
Update index.js

pgrobban commented on index.js:
```diff
@@ -0,0 +1,60 @@
+    
+    let finalCheck = isAlphabetical(inOrder)
+
+    if (finalCheck === true){
```
you don't need the === true part.

pgrobban commented on index.js:
```diff
@@ -0,0 +1,60 @@
+        return false
+    }
+
+    return join
```
don't mix return types. it will be difficult for the caller of the function to keep track of what it returns.

pgrobban commented on index.js:
```diff
@@ -0,0 +1,60 @@
+
+        if (arr[i+1] > arr[i]){
+            count += 1
+        }}
```
the second brace should be on a new line since it closes the for. please run a format (alt+shift+F) before submitting new files.

pgrobban commented on index.js:
```diff
@@ -0,0 +1,60 @@
+            count += 1
+        }}
+
+    if (count === arr.length-1){ 
```
instead of the if you can return the value directly
```return count === arr.length - 1```
since this is a boolean already.

henyspider committed:
Update index.js

henyspider committed:
Update index.js

pgrobban commented on index.js:
```diff
@@ -0,0 +1,58 @@
+    
+    let finalCheck = isAlphabetical(inOrder)
+
+    if (finalCheck){
```
why not return finalCheck directly? since you want to return a boolean
you don't even need to create the variable.
return isAlphabetical(inOrder)

pgrobban commented on index.js:
```diff
@@ -0,0 +1,58 @@
+    return true
+}
+
+const isAlphabetical = arr => {
```
to be consistent make this a normal function call
```suggestion
function isAlpabetical(arr) {
```

and to be nitpicky maybe call this function ```isArrayAlphabetical``` since it's not working on strings

henyspider committed:
Update index.js

Co-authored-by: Robert Sebescen <pgrobban@gmail.com>

henyspider committed:
Update index.js

henyspider committed:
New Changes

Here are the changes I made. I understood what you said however I didn't know how to modify my code the way you wanted me to, it was a little confusing as to where to put what etc. So I took in consideration the last messages you sent me, where you said "one function for.. one function for... one function for..." and I came up with this. I'm not sure this is exactly what you wanted me to do, again I understood the three-functions-code part but I didn't really know how to modify my code to make it fit, so I'm hoping this is what you expected me to do haha, at least a little bit, otherwise I don't really know..

pgrobban commented:
Two general comments:
Please always format your files before doing a PR.
Make sure your commits have useful messages like "changed function x to return a string" or something instead of just "update" and "new changes". It will be useful for you to if you have to go back in history.

pgrobban commented on index.js:
```diff
@@ -0,0 +1,59 @@
+}
+
+
+function isLongestWord(arr){
```
functions called isXXX or hasXXX are expected to return booleans.
This function returns the longest word so i would call it getLongestWord.

pgrobban commented on index.js:
```diff
@@ -0,0 +1,59 @@
+        }
+    }
+
+return longest
```
please format the file, the indentation here is weird.

pgrobban commented on index.js:
```diff
@@ -0,0 +1,59 @@
+const words = data.split('\r\n');
+const ALPHABET = ['a', 'b', 'c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z'];
+
+function areAlphabetical(words){
```
if i want to know what a function does quickly i should be able to tell from the name. this function is called areAlphabetical but it's not telling me if words is alphabetical, What it does is returning the longest alphabetical word.
So I would call this "getLongestAlphabeticalWord".

henyspider committed:
Update index.js

henyspider committed:
Modified the function passed in the console.log

I changed the name of the first function, which i console.logged, and forgot to change the name of that same function line 59.
